src/macros-by-example.md

```diff
@@ -24,149 +24,475 @@
-together.
-
-The macro expander matches and transcribes every token that does not begin with
-a `$` literally, including delimiters. For parsing reasons, delimiters must be
-balanced, but they are otherwise not special.
-
-In the matcher, `$` _name_ `:` _designator_ matches the nonterminal in the Rust
-syntax named by _designator_. Valid designators are:
-
-* `item`: an [_Item_]
-* `block`: a [_BlockExpression_]
-* `stmt`: a [_Statement_] without the trailing semicolon
-* `pat`: a [_Pattern_]
-* `expr`: an [_Expression_]
-* `ty`: a [_Type_]
-* `ident`: an [IDENTIFIER_OR_KEYWORD]
-* `path`: a [_TypePath_] style path
-* `tt`: a [_TokenTree_]&nbsp;(a single [token] or tokens in matching delimiters `()`, `[]`, or `{}`)
-* `meta`: a [_MetaItem_], the contents of an attribute
-* `lifetime`: a [LIFETIME_TOKEN]
-* `vis`: a [_Visibility_] qualifier
-* `literal`: matches `-`<sup>?</sup>[_LiteralExpression_]
+Each macro by example has a name, and one or more _rules_. Each rule has two
+parts: a _matcher_, describing the syntax that it matches, and a _transcriber_,
+describing the syntax that will replace a successfully matched invocation. Both
+the matcher and the transcriber must be surrounded by delimiters. Macros can
+expand to expressions, statements, items (including traits, impls, and foreign
+items), types, or patterns.
+
+## Transcription
+
+When a macro is invoked, the macro expander looks up macro invocations by name,
+and tries each macro rule in turn. It transcribes the first successful match; if
+this results in an error, then future matches are not tried. When matching, no
+lookahead is performed; if the compiler cannot unambiguously determine how to
+parse the macro invocation one token at a time, then it is an error. In the
+following example, the compiler does not look ahead past the identifier to see
+if the following token is a `)`, even though that would allow it to parse the
+invocation unambiguously:
+
+```rust,compile_fail
+macro_rules! ambiguity {
+    ($($i:ident)* $j:ident) => { };
+}
+
+ambiguity!(error); // Error: local ambiguity
+```
+
+In both the matcher and the transcriber, the `$` token is used to invoke special
+behaviours from the macro engine (described below in [Metavariables] and
+[Repetitions]). Tokens that aren't part of such an invocation are matched and
+transcribed literally, with one exception. The exception is that the outer
+delimiters for the matcher will match any pair of delimiters. Thus, for
+instance, the matcher `(())` will match `{()}` but not `{{}}`. The character
+`$` cannot be matched or transcribed literally.
+
+When forwarding a matched fragment to another macro-by-example, matchers in
+the second macro will see an opaque AST of the fragment type. The second macro
+can't use literal tokens to match the fragments in the matcher, only a
+fragment specifier of the same type. The `ident`, `lifetime`, and `tt`
+fragment types are an exception, and can be matched by literal tokens. The
```

Comment:
Just wanted to make sure you intentionally removed the emphasis on “can” here.

Reply:
Nope, was a copy/paste mistake.  Wish it was easier to grab the raw markdown in github.